### PR TITLE
don't allow float for slicing

### PR DESCRIPTION
**Pull request by godofdream**
don't allow float for slicing, and making sure negative indexes work as intended

### CODE REVIEW

File: src/builtins/filters/array.rs [hunk 1]
```diff
@@ -244,45 +244,49 @@ pub fn map(value: &Value, args: &HashMap<String, Value>) -> Result<Value> {
...
-    if end > arr.len() {
-        end = arr.len();
+    if end > len {
+        end = len;
     }
-
+    if end < 0 && len + end >= 0 {
+        end += len;
+    }
+    println!("{} {}", start, end);
```
Comment (Keats) [inline src/builtins/filters/array.rs:282]: should be removed

File: src/builtins/filters/array.rs [hunk 2]
```diff
@@ -244,45 +244,49 @@ pub fn map(value: &Value, args: &HashMap<String, Value>) -> Result<Value> {
...
     // Not an error, but returns an empty Vec
-    if start >= end {
+    if start >= end || end < 0 {
         return Ok(Vec::<Value>::new().into());
     }
-
-    Ok(arr[start..end].into())
+    #[allow(clippy::cast_possible_truncation)]
+    #[allow(clippy::cast_sign_loss)]
```
Comment (Keats) [inline src/builtins/filters/array.rs:288]: what does clippy suggest to do?

File: src/builtins/filters/array.rs [hunk 3]
```diff
@@ -244,45 +244,49 @@ pub fn map(value: &Value, args: &HashMap<String, Value>) -> Result<Value> {
...
     if arr.is_empty() {
         return Ok(arr.into());
     }
 
-    let start = match args.get("start") {
-        Some(val) => get_index(try_get_value!("slice", "start", f64, val), &arr),
+    let len = arr.len() as i64;
+
+    let mut start = match args.get("start") {
+        Some(val) => try_get_value!("slice", "start", i64, val),
```
Comment (Keats) [inline src/builtins/filters/array.rs:260]: sounds like a breaking change if someone was passing a float

**Review by Keats — CHANGES_REQUESTED** [review]

File: src/builtins/filters/array.rs (hunk 3 above)
Comment (godofdream) [reply src/builtins/filters/array.rs:260]: Float wasn't documented.
I will add a fallback for floats.

Comment (godofdream) [reply src/builtins/filters/array.rs:260]: If we want to keep floats as Index (what seems like a bad Idea, but is understandable for backwards compatibility) we can close this MR

**Comment by godofdream** [comment]
As we want to keep floats as Index (what seems like a bad Idea, but is understandable for backwards compatibility) we close this MR

**Pull request closed without merging**